### PR TITLE
fix: do not emit empty templatized args.

### LINKED ISSUE #123: Uint8Array < > (without type arguments)
`Uint8Array` should be a non-templatized subtype of the templatized type `ArrayLike<number>`.

We currently generate `Uint8Array < >` for it.

### PR DESCRIPTION
**Pull request by mprobst**
In Closure, types that filled all template args from a super type are
still represented by a templatized type.

Fixes #123.

### CODE REVIEW

File: src/test/java/com/google/javascript/clutz/empty_type_args.d.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+declare namespace ಠ_ಠ.clutz_internal.empty_type_args {
+  interface ITemplated < T > {
+  }
+}
+declare module 'goog:empty_type_args.ITemplated' {
+  import alias = ಠ_ಠ.clutz_internal.empty_type_args.ITemplated;
+  export default alias;
+}
+declare namespace ಠ_ಠ.clutz_internal.empty_type_args {
+  class NoMoreTemplateArgs implements ITemplated < number > {
```
Comment (alexeagle) [inline src/test/java/com/google/javascript/clutz/empty_type_args.d.ts]: I don't understand which line here was absent before the fix. Was there no `<number>` here?

Comment (mprobst) [inline src/test/java/com/google/javascript/clutz/empty_type_args.d.ts]: Yeah, our golden masters are getting a bit wordy.

This is all just part of the buildup - the problem was that when `NoMoreTemplateArgs` was used, it was printed as `NoMoreTemplateArgs < >`, which is bad syntax in TS.
